src/integration/react.js

```diff
@@ -51,16 +51,27 @@ export class PersistGate extends PureComponent<Props, State> {
   }
 
   render() {
+    const { children, loading } = this.props
+
     if (process.env.NODE_ENV !== 'production') {
-      if (typeof this.props.children === 'function' && this.props.loading)
+      if (typeof children === 'function' && loading)
         console.error(
           'redux-persist: PersistGate expects either a function child or loading prop, but not both. The loading prop will be ignored.'
         )
     }
-    if (typeof this.props.children === 'function') {
-      return this.props.children(this.state.bootstrapped)
+    if (typeof children === 'function') {
+      return children(this.state.bootstrapped)
     }
 
-    return this.state.bootstrapped ? this.props.children : this.props.loading
+    if (this.state.bootstrapped) {
+      if (children) {
+        return children
+      }
+      return null
```

Comment:
This surely works! But in my opinion, this is what defaultProps should be used for.
You get around removing the `?` but it feels just like another workaround :D 
But the way flow handles defaultProps is to blame in the end.

Reply:
Yes I totally agree with you, I did mention that in the `important comment` section on the description, this should be Facebook's Flow work, not to be done here.

I'll try to take a look on Facebook's Flow repo and I might be able to make a good contribution.